controllers/instance_controller.go

```diff
@@ -113,25 +113,36 @@ func (r *InstanceReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return ctrl.Result{}, err
 	}
 
-	r.Log.Info("Migrating state")
-
+	r.Log.Info("migrating state")
 	if err := ops.MigrateState(ctx); err != nil {
 		progressing.Reason = "MigrateFailed"
 		progressing.Message = err.Error()
 		return ctrl.Result{}, err
 	}
 
-	r.Log.Info("Rendering values")
-
+	r.Log.Info("rendering values")
 	objects, err := ops.Build(ctx)
 	if err != nil {
 		progressing.Reason = "BuildFailed"
 		progressing.Message = err.Error()
 		return ctrl.Result{}, err
 	}
 
-	r.Log.Info("Applying resources")
+	r.Log.Info("finishing previous deployment")
+	result, err := ops.FinishDeployment(ctx)
+	if err != nil {
+		progressing.Reason = "PreviousDeploymentFailed"
+		progressing.Message = err.Error()
+		return ctrl.Result{}, err
+	}
```

Comment:
vague thought: something seems weird about building the next state, then finalising the previous state, then applying the next state. Am I misunderstanding, or is there some way to have this be "finish previous state, prepare next state, apply next state"?